harmonica/filters/_filters.py

```diff
@@ -62,3 +62,598 @@ def derivative_upward_kernel(fft_grid, order=1):
+    frequency coordinates of the passed ``fft_grid`` and is defined as follows:
+
+    .. math::
+
+            g(\mathbf{k}) = \frac{149.8}{f|\mathbf{k}|}\times
+            {reduction\_to\_pole\_kernel}
+
+
+    where :math:`\mathbf{k}` is the wavenumber vector
+    (:math:`\mathbf{k} = 2\pi \mathbf{f}`
+
+    Parameters
+    ----------
+    fft_grid : :class:`xarray.DataArray`
+        Total Magnetic Intensity or Reduction To Pole magnetic field.
+        Please set inclination = 90 for Reduction To Pole magnetic field.
+        Array with the Fourier transform of the original grid.
+        Its dimensions should be in the following order:
+        *freq_northing*, *freq_easting*.
+        Use :func:`xrft.xrft.fft` and :func:`xrft.xrft.ifft` functions to
+        compute the Fourier Transform and its inverse, respectively.
+    inclination : float in degrees
+        The inclination inducing Geomagnetic field. Default is 90 degree for
+        RTP field.
+    declination : float in degrees
+        The declination inducing Geomagnetic field. Default is 0 degree for
+        RTP field.
+    magnetization_inclination : float in degrees
+        The inclination of the total magnetization of the anomaly source.
+        Default is i, neglecting remanent magnetization and
+        self demagnetization.
+    magnetization_declination : float in degrees
+        The declination of the total magnetization of the anomaly source.
+        Default is d, neglecting remanent magnetization and
+        self demagnetization.
+    f : float or 2d-array
+        Ambient field in the study area. It can use the mean ambinent field
+        value in the study area or the real ambient field value in all
+        locations. Default is 50,000 nT.
+
+    Returns
+    -------
+    da_filter : :class:`xarray.DataArray`
+        Array with the kernel for the pseudo gravity filter in frequency
+        domain.
+
+    References
+    ----------
+    [Salem2014]_
+
+    See also
+    --------
+    harmonica.pseudo_gravity
+    """
+    # Check if magnetization angles are valid
+    _check_magnetization_angles(magnetization_inclination, magnetization_declination)
+    # Define magnetization angles if they are None
+    if magnetization_declination is None and magnetization_inclination is None:
+        magnetization_inclination = inclination
+        magnetization_declination = declination
+    # Catch the dims of the Fourier transformed grid
+    dims = fft_grid.dims
+    # Grab the coordinates of the Fourier transformed grid
+    freq_easting = fft_grid.coords[dims[1]]
+    freq_northing = fft_grid.coords[dims[0]]
+    # Convert frequencies to wavenumbers
+    k_easting = 2 * np.pi * freq_easting
+    k_northing = 2 * np.pi * freq_northing
+    # Check if input is RTP field
+    if inclination == 90:
+        # Calculate vertical integral
+        da_filter = np.sqrt(k_easting**2 + k_northing**2) ** -1
```

Comment:
What happens if `inclination == -90`? Shouldn't we consider it as a separate case as well?

Reply:
Got it! Makes sense then.